pkg/network/setup/netconf.go

```diff
@@ -34,9 +36,11 @@ type cacheCreator interface {
 }
 
 type NetConf struct {
-	setupCompleted sync.Map
-	cacheCreator   cacheCreator
-	nsFactory      nsFactory
+	setupCompleted   sync.Map
+	cacheCreator     cacheCreator
+	nsFactory        nsFactory
+	configStateCache map[types.UID]ConfigStateCache
```

Comment:
Can we place here the higher level `ConfigState` object?
This will hide the inner cache and make it an implementation detail to the rest of the code.

nit: Is it a must to use `types.UID`? I think a string would have also worked, avoiding this special import.